jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/system/spot/SpotTDB1.java

```diff
@@ -0,0 +1,234 @@
+     *
+     * primaryIndexPrefix     dft:GPU
+     * prefixIndexes          dft:[GPU]
+     * -- Actual name of GPU index.
+     * indexPrefix            dft:prefixIdx BPT
+     *
+     * -- NodeTable
+     * indexNode2Id           dft:node2id
+     * indexId2Node           dft:nodes
+     *
+     * -- NodeTable
+     * prefixNode2Id          dft:prefix2id BPT *
+     * prefixId2Node          dft:prefixes DAT *
+     */
+
+    /**
+     * Test to see is a location is either empty (and a fresh TDB1 database can be
+     * created there) or has looks like it is an existing TDB1 database. See
+     * {@link #checkTDB1(Location)} for a test that the location is a valid, existing
+     * database.
+     */
+    public static boolean isTDB1(String pathname) {
+        return isTDB1(Location.create(pathname));
+    }
+
+    /**
+     * Test to see is a location is either empty (and a fresh TDB1 database can be
+     * created there) or has looks like it is an existing TDB1 database. See
+     * {@link #checkTDB1(Location)} for a test that the location is a valid, existing
+     * database.
+     */
+    public static boolean isTDB1(Location location) {
+        if ( location.isMem() )
+            return true;
+
+        if ( ! location.exists() )
+            return false;
+        if ( isEmpty(location) )
+            return true;
+
+        // Very occasionally, deleting the journal makes sense.
+
+        return isTDB1_Storage(location);
+    }
+
+    /** Quick check for TDB1. Does not check for validity. */
+    private static boolean isTDB1_Storage(Location location) {
+        // Has a journal.
+        //return exists(location, Names.journalFileBase, Names.extJournal);
+        // Test for triples primary index
+        StoreParams params = getStoreParams(location);
+        return exists(location, params.getPrimaryIndexTriples(), Names.bptExtTree, Names.bptExtRecords);
+    }
+
+    /**
+     * Check all files exist for a TDB1 database, or the area is empty (and so a new
+     * database can be created in the location). Throw an exception if a file
+     * is missing.
+     */
+    public static void checkTDB1(String pathname) {
+        checkTDB1(Location.create(pathname));
+    }
+
+    /**
+     * Check all files exist for a TDB1 database, or the area is empty (and so a new
+     * database can be created in the location). Throw an exception if a file
+     * is missing.
+     */
+    public static void checkTDB1(Location location) {
+        if ( location.isMem() )
+            return;
+
+        if ( isEmpty(location) )
+            return;
+
+        // Journal, fixed name.
+        validate(location, Names.journalFileBase, Names.extJournal);
+
+        if ( ! isTDB1(location) )
+            throw new TDBException("Not a TDB1 location: "+location);
+
+        // Places for StoreParams: location or default
+        StoreParams params = getStoreParams(location);
+
+        validate(location, Names.journalFileBase, Names.extJournal);
```

Comment:
I think this is being validated a few lines above already?

Reply:
Fixed